manager/integration/tests/backupstore.py

```diff
@@ -0,0 +1,410 @@
+        return nfs_corrupt_backup_cfg_file()
+
+
+# TODO: implement nfs_corrupt_backup_cfg_file
+def nfs_corrupt_backup_cfg_file():
+    raise NotImplementedError
+
+
+def minio_corrupt_backup_cfg_file(client, core_api, volume_name, backup_name): # NOQA
+    corrupt_backup_cfg_data = "{corrupt: definitely"
+
+    backupstore_write_backup_cfg_file(client,
+                                      core_api,
+                                      volume_name,
+                                      backup_name,
+                                      corrupt_backup_cfg_data)
+
+
+def backupstore_delete_dummy_in_progress_backup(client, core_api, volume_name):
+    backupstore_delete_backup_cfg_file(client,
+                                       core_api,
+                                       volume_name,
+                                       "backup-dummy")
+
+
+def backupstore_delete_random_backup_block(client, core_api, volume_name):
+    backupstore = backupstore_get_backup_target(client)
+
+    if is_backupTarget_s3(backupstore):
+        minio_delete_random_backup_block(client, core_api, volume_name)
+
+    elif is_backupTarget_nfs(backupstore):
+        nfs_delete_random_backup_block()
+
+
+# TODO: implement nfs_delete_random_backup_block
+def nfs_delete_random_backup_block():
+    raise NotImplementedError
+
+
+def minio_delete_random_backup_block(client, core_api, volume_name):
+    secret_name = backupstore_get_secret(client)
+    assert secret_name != ''
+
+    minio_api = minio_get_api_client(client, core_api, secret_name)
+
+    bucket_name = minio_get_backupstore_bucket_name(client)
+    backup_blocks_dir = minio_get_backup_blocks_dir(volume_name)
+
+    block_object_files = minio_api.list_objects(bucket_name,
+                                                prefix=backup_blocks_dir,
+                                                recursive=True)
+
+    object_file = block_object_files.__next__().object_name
+
+    try:
+        minio_api.remove_object(bucket_name, object_file)
+    except ResponseError as err:
+        print(err)
+
+
+def backupstore_count_backup_block_files(client, core_api, volume_name):
+    backupstore = backupstore_get_backup_target(client)
+
+    if is_backupTarget_s3(backupstore):
+        return minio_count_backup_block_files(client, core_api, volume_name)
+
+    elif is_backupTarget_nfs(backupstore):
+        return nfs_count_backup_block_files()
+
+
+# TODO: implement nfs_count_backup_block_files
+def nfs_count_backup_block_files():
+    raise NotImplementedError
+
+
+def minio_count_backup_block_files(client, core_api, volume_name):
+    secret_name = backupstore_get_secret(client)
+    assert secret_name != ''
+
+    minio_api = minio_get_api_client(client, core_api, secret_name)
+    bucket_name = minio_get_backupstore_bucket_name(client)
+    backup_blocks_dir = minio_get_backup_blocks_dir(volume_name)
+
+    block_object_files = minio_api.list_objects(bucket_name,
```

Comment:
have a minio_list_files method, then just pass the blocks prefix.
same for the delete random block -> list then pick one of the prefixes call delete_file

Reply:
fixed